episodes/04-muscle-couple.md

```diff
@@ -26,20 +26,46 @@ We managed to connect one of our sub-models, the reaction model, to MUSCLE. In o
 ## Investigating the macro model
 
 
-## Connecting it all together
-With both models defined, we now need to instruct MUSCLE3 on how to connect them together. We do this by creating an object of type ymmsl.Configuration, which contains all the information needed to run the simulation. Remember the diagram we drew in episode 2: 
+## Connecting the models together
 
-![Diagram to be added](/fig/ep01-plane-model-coupling.png){alt='alt text for accessibility purposes'}
+With both models defined, we now need to instruct MUSCLE3 on how to connect them together. To find out how to connect the models, we first need to apply the MMSF to this particular simulation. The reaction and diffusion processes act simultaneously on the same spatial and temporal domain. The discretisation of that domain in space is also the same for the two models, so that they have the same spatial scale. As mentioned before, we're going to configure the models to have temporal scale separation, with the diffusion model the slow macromodel and the reaction model the fast micromodel.
 
-It shows that there are two components named macro and micro. A conduit connects port state_out on macro to state_in on micro. The symbols at the ends of the conduit show the operators that the ports belong to, O_I for macro.state_out and F_INIT for micro.state_in. Another conduit connects port micro.final_state (O_F) to macro.state_in (S).
+According to the MMSF, this means we have to use the Call-and-Release coupling template, with one instance of each submodel:
 
-Note that there’s a mnemonic here: Operators O_I and S, which are within the state update loop, have a circular symbol, while F_INIT and O_F use a diamond shape. Also, filled symbols designate ports on which messages are sent, while open symbols designate receiving ports. We can therefore see that for each state update, macro will send on state_out, after which micro will do a full run and send its final result as input for the next state update of macro.
+![gMMSL diagram for the reaction-diffusion model](fig/ep04-reaction-diffusion-coupling.png){alt='gMMSL diagram for the reaction-diffusion model. Two boxes labeled macro and micro represent the two submodels. A line connects a filled circle labeled state_out on macro to an open diamond labeled state_in on micro. A second line connects a filled diamond labeled final_state on micro to an open circle labeled state_in on macro.'}
```

Comment:
Actually, if this example is not mentioned in the theory section, I think we already need this diagram in Episode 3, to know which operators we need to couple to which ports, right?

Reply:
Well, in principle you shouldn't need to think about how the model will be coupled when you make it, but in practice you do because it saves time. To do this really nicely, both models should receive the state on f_init and S, and send it on Oi and Of, and provide the current state as a default on the S receive so that the model can be used without a micromodel attached.

I think we can just leave this as a isn't-it-nice-how-this-works-out, it's a teaching example after all. Maybe we can have a callout somewhere with this explanation?